File: Python/8py 3/period_is_late.py
"""
|--------------------------------------------------------------------------
| Is your period late?
|--------------------------------------------------------------------------
|
"""

def period_is_late(last, today, cycle_length):
   return (today - last).days > cycle_length


def period_is_late(last, today, cycle_length):
   return today.toordinal() - last.toordinal() > cycle_length


from datetime import timedelta
def period_is_late(last,today,cycle_length):
   return today > (last + timedelta(days=cycle_length))

File: PHP/8php 2/periodIsLate.php
<?php
/*
|--------------------------------------------------------------------------
| Is your period late?
|--------------------------------------------------------------------------
|
*/

function periodIsLate($last, $today, $cycleLength) {
   return ($today->diff($last)->days > $cycleLength);
}


function periodIsLate($last, $today, $cycleLength) {
   return ($today->getTimestamp() - $last->getTimestamp()) / (60 * 60 * 24) > $cycleLength;
}


function periodIsLate($last, $today, $cycleLength) {
   return $last->add(new DateInterval("P{$cycleLength}D")) < $today;
}